### PR TITLE
first apss rest of the specs stuff

### PR DESCRIPTION
**Pull request by paetling**
WIP

### CODE REVIEW

File: docs/specs/test-specs.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Dusty Testing Specs
+These specs are used to specify tests you can run in apps and libs. These do not exist as distinct documents but as subfields of lib and app specs. They are found under the **test** key.
+
+## Keys
+
+### build and image
+The image and build keys in test specs mirror the image and build keys in the [app-spec](./app-specs.md#build).
```
Comment (jsingle) [inline docs/specs/test-specs.md]: Maybe note how test containers are different from run containers?

File: docs/specs/test-specs.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  - testRedis
+```
+The services key represents a list of services that the app or lib being tested depends on. These services will be spun up and daisy chained together over the Docker network bridge.  This is done using Docker Compose's **net** key.
+
+### once
+```
+once:
+  - pip install -r test_requirements.txt
+```
+The once key is a list of commands to be run when you are creating the testing base image.  In order to speed up the process of running tests, the first time tests are run, Dusty will create a testing base image based off of the image or build key and the once commands. This image is then used to run the actual tests.  This allows the heavy install commands to only happen the very first time tests are run. From there forward, tests should run very fast. <br />
```
Comment (jsingle) [inline docs/specs/test-specs.md]: Last sentence is a little awk

File: docs/specs/test-specs.md (hunk 1 above)
Comment (paetling) [inline docs/specs/test-specs.md]: do some noting of that with the once command and the services command. Let me know if you think i need more or should consolidate them somewhere.

**Comment by thieman** [comment]
This is all a good start. Once the shell's in place, it's a lot easier to go through and make targeted improvements. Merge that shit.

**Pull request merged**